python/feldera/rest/feldera_client.py

```diff
@@ -671,14 +675,62 @@ def push_to_pipeline(
             content_type = "text/csv"
             data = bytes(str(data), "utf-8")
 
-        self.http.post(
+        resp = self.http.post(
             path=f"/pipelines/{pipeline_name}/ingress/{table_name}",
             params=params,
             content_type=content_type,
             body=data,
             serialize=serialize,
         )
 
+        token = resp.get("token")
+        if token is None:
+            raise FelderaAPIError("response did not contain a completion token", resp)
+
+        if not wait:
+            return token
+
+        self.wait_for_token(pipeline_name, token)
+
+        return token
+
+    def wait_for_token(self, pipeline_name: str, token: str):
+        """
+        Blocks until this all records represented by this completion token have
+        been processed.
+
+        :param pipeline_name: The name of the pipeline
+        :param token: The token to check for completion
+        """
+
+        params = {
+            "token": token,
+        }
+
+        start = time.time()
+
+        while True:
+            resp = self.http.get(
+                path=f"/pipelines/{pipeline_name}/completion_status", params=params
+            )
+
+            status: Optional[str] = resp.get("status")
+            if status is None:
+                raise FelderaAPIError(
+                    f"got empty status when checking for completion status for token: {token}",
+                    resp,
+                )
+
+            if status.lower() == "complete":
+                break
+
+            elapsed = time.time() - start
+            logging.debug(
+                f"still waiting for inputs represented by {token} to be processed; elapsed: {elapsed}s"
+            )
+
+            time.sleep(0.1)

```

Comment:
That's really fast polling. I guess that in the common case, records will be completed quickly, which is an argument for polling frequently. But it's a lot if it takes a long time.

Sometimes, in cases like this, I've set up polling so that it's fast at first but slows down after a while.